src/Service/RequestService.php

```diff
@@ -826,16 +927,22 @@ private function getConfigInOutOrGlobal(string $endpointRef, string $endpoint, a
         foreach (['in', 'out'] as $type) {
             if (array_key_exists($endpointRef, $applicationConfig) === true && array_key_exists($type, $applicationConfig[$endpointRef])) {
                 $appEndpointConfig[$type] = $applicationConfig[$endpointRef][$type];
-            } elseif (array_key_exists($endpoint, $applicationConfig) === true && array_key_exists($type, $applicationConfig[$endpoint])) {
+                continue;
+            }
+
+            if (array_key_exists($endpoint, $applicationConfig) === true && array_key_exists($type, $applicationConfig[$endpoint])) {
                 $appEndpointConfig[$type] = $applicationConfig[$endpoint][$type];
-            } elseif (array_key_exists('global', $applicationConfig) === true && array_key_exists($type, $applicationConfig['global'])) {
+                continue;
+            }
+
+            if (array_key_exists('global', $applicationConfig) === true && array_key_exists($type, $applicationConfig['global'])) {
                 // Do global last, so that we allow overwriting the global options for specific endpoints ^.
                 $appEndpointConfig[$type] = $applicationConfig['global'][$type];
+                continue;
             }
         }
```

Comment:
I guess else if here still might be desirable, but I'm a bit struggling to understand this.